src/System.Security.Cryptography.X509Certificates/tests/ChainTests.cs

```diff
@@ -63,6 +63,104 @@ public static void BuildChain()
             }
         }
 
+#if netcoreapp11
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [Fact]
+        public static void VerifyChainFromHandle()
+        {
+            X509Chain chain;
+
+            using (var microsoftDotCom = new X509Certificate2(TestData.MicrosoftDotComSslCertBytes))
+            using (var microsoftDotComIssuer = new X509Certificate2(TestData.MicrosoftDotComIssuerBytes))
+            using (var microsoftDotComRoot = new X509Certificate2(TestData.MicrosoftDotComRootBytes))
+            using (var unrelated = new X509Certificate2(TestData.DssCer))
+            using (var chainHolder = new ChainHolder())
+            {
+                chain = chainHolder.Chain;
+
+                chain.ChainPolicy.ExtraStore.Add(unrelated);
+                chain.ChainPolicy.ExtraStore.Add(microsoftDotComRoot);
+                chain.ChainPolicy.ExtraStore.Add(microsoftDotComIssuer);
+                chain.ChainPolicy.VerificationFlags = X509VerificationFlags.AllowUnknownCertificateAuthority;
+                chain.ChainPolicy.VerificationTime = new DateTime(2015, 10, 15, 12, 01, 01, DateTimeKind.Local);
+                chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
+
+                bool valid = chain.Build(microsoftDotCom);
+                Assert.True(valid, "Chain built validly");
+
+                X509Chain chain2 = new X509Chain(chain.ChainContext);
```

Comment:
If the certs from chain2.ChainElements are distinct objects from chain.ChainElements they need to be Disposed. (make a new ChainHolder over this chain).

If they're not distinct elements, I guess they don't.
But really a test should exist that Asserts their Sameness (or NotSameness)